Review installation script

slobberbone opened the pull request:
# Principe
Le script principal postinst a été remplacé par les scripts suivants selon le cas dans lequel vous vous trouvez : 
- install_git.sh
- install_git_dev.sh
- install_docker.sh

Le postinst a été découpé en 2 scripts dans /install/scripts/
- preinst.sh
- posinst.sh

Le .deb lui n'aura qu'à appeler le posinst.sh comme avant. @Astral0 

# Particularités
## Sous scripts
Les nouveau scripts incluent : 
- un fichier de commandes : utils.sh
- un fichier de configuration : config.sh
  - Ce dernier est à modifier pour gérer les cas des installations multi serveurs/conteneurs, environnements de dev, ...
  - Si vous n'avez pas les sources dnas /usr/share/nextdom, le script devrait surcharger le chemin d'installation défini ROOT_DIRECTORY

# Tests & Validations
Voici les différents cas de tests
Vous devez mettre quel contexte vous avez validé en commentaire de la PR, Je tiendrai la liste des contextes validés à jour

## Installation sur une plateforme vierge

### Fresh install VM Debian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install VM Debian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [x] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install LXC Debian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install LXC Debian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement 
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install Docker Debian 9###  (ne pas utiliser le script install_docker)
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install Docker Debian 10###  (ne pas utiliser le script install_docker)
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install Docker sous Windows Debian 10###  (ne pas utiliser le script install_docker)
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install ARMbian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [x] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install ARMbian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement 
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [x] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install Raspbian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Fresh install Raspbian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql


## Migration d'une installation existante

### Migration VM Debian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration VM Debian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration LXC Debian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration LXC Debian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement 
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration Docker Debian 9###  (ne pas utiliser le script install_docker)
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration Docker Debian 10###  (ne pas utiliser le script install_docker)
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration Docker sous Windows 10 Debian 10###  (ne pas utiliser le script install_docker)
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration ARMbian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration ARMbian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement 
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration Raspbian 9
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

### Migration Raspbian 10
- Sources dans /usr/share/nextdom
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql
  - [ ] Installation par repository debian uniquement
- Sources dans /ouvousvoulez
  - [ ] Installation par Git
  - [ ] Installation par Git + paquets debian : nextdom-common et nextdom-mysql

Esso1-jeedom commented:
![image](https://user-images.githubusercontent.com/40739563/71307769-64007c80-23f3-11ea-907d-98e0e5324ea0.png)

Esso1-jeedom commented:
install git classique 
ou j'ai fait un changement de branche
install_git.sh
var/www/html/install#

edgd1er commented:
Large remaniement ;) je n ai pas testé le code, juste lu. 
SysVInit à été remplacé par SystemD, pourquoi ne pas utiliser systemctl ? 
qq remarques pr docker principalement :
Est il possible de detecter l etat docker pour bypasser les services stop/start, inutiles ds docker ?
le conteneur mysql aura deja son user, mdp et db crees. Est il possible de bypasser les étapes de création si le mdp root mysql est absent/vide ?

slobberbone committed:
Review installation script


Add preinst file


Review installation script


Add preinst file


preinst


postinst


add preinst


avancement


test


Some Fixes

edgd1er commented:
pour les addlog, en mode docker aucun dossier n'est préexistant, du coup addLogInfo plante dès le début car /var/log/nextdom n'existe pas, il faudrait le créer et/ou detecter son absence.

edgd1er committed:
Proposition pour install docker
- variables d'environnement prioritaire sur les variables des scripts
- chargement des fonctions et construction à la volée du script
- ajout d'un bypass pour les fonctions inutiles pour docker(service)
- Nextdom_secret non géré pour le moment (quel est son utilité ? doit il etre parametrable via une variable d'environnement)

edgd1er committed:
ajout du supervisor pour docker / gestion production - dev

slobberbone commented:
@edgd1er oui pour le moment c'était voulu de ne proposer "que" la même chose qu'avant pour permettre de ne pas constater de régressions.
Pour tes modifications, je veux bien les prendre mais le soucis, c'est que tu as archivé des modifications en mm temps que des tabulations supprimées ou rajoutées ... du coup c pas très lisible

slobberbone committed:
Harmonize + BDD changes

slobberbone commented:
@edgd1er j'ai gardé ce que tu avais fais, mais virer les options ${docker} en direct mais j'ai rajouté dans le config.sh
Là tout semble ok

slobberbone commented:
@edgd1er Pour tes améliorations, je te propose de voir ça dans un second temps

Esso1-jeedom commented:
![image](https://user-images.githubusercontent.com/40739563/71957777-edcf8880-31ee-11ea-8a8a-725adb574795.png)

Esso1-jeedom commented:
![image](https://user-images.githubusercontent.com/40739563/71958896-b6aea680-31f1-11ea-9eb5-2b4e7bf1eb2d.png)

slobberbone reviewed (commented):
c bizare @Esso1-jeedom ca donne l'impression qu'il te manque des fichiers ...

Gis70 reviewed (commented):
![image](https://user-images.githubusercontent.com/16589132/72020871-09797400-326d-11ea-9786-970cb812fd2a.png)

dossier /html renommé

Peux tu écrire la procédure pour une installation depuis github

slobberbone committed:
Change bdd nextdomdev to nextdom

vinceg77 commented:
@testeurFou : tester 
1/ sur différentes plateformes
2/ Depuis une install vierge
3/ Depuis un update (de 0.5.4 à 0.6.0 par exemple)

Esso1-jeedom commented:
alors c toujours pareille pour moi . aprés une install Git classique . je passe sur la branche ,
je lance le install git.. et rien ne marche .

Esso1-jeedom commented:
bon aprés modif  du config.sh 
ROOT_DIRECTORY="/var/www/nextdomdev"

![image](https://user-images.githubusercontent.com/40739563/72143231-9d932a80-3396-11ea-8c82-61040df40e53.png)

ca marche pour moi

Gis70 reviewed (changes requested):
- Partant d'une install sans nextdom, jai dû lancer gen_composer manuellement avec de lancer l'install git, pourquoi n'est il pas lancé ?

- De plus il y a un soucis de gen_assets aussi ? 

![image](https://user-images.githubusercontent.com/16589132/72193463-e0421a80-3409-11ea-8af3-584a49fc604e.png)

- Pb d'affichage du mot de passe de la db ?

![image](https://user-images.githubusercontent.com/16589132/72193505-2303f280-340a-11ea-8116-d6f063a249a5.png)

edgd1er commented:
Plusieurs remarques:

* pourquoi remplacer les tests []  (compatible posix) par des [[ ]] ( non compatibles posix) alors que ce n'est pas nécéssaire.
* pourquoi mettre la configuration mysql a deux endroits maintenant config.sh et mysql/secret ? Pour docker, la conf est a 4 endroits maintenant: docker-compose.yml, mysql/secret, config.sh, common.config.php. quelle est l'utilité de ce mysql secret ? il n'est pas spécifique docker ( besoin de swarm)
* aucune fonction ne modifie "result", de ce fait le log indique a chaque fois que c'est ok
* les booleen avec bash sont "bizarres", [[ ! $VARIABLE ]] sera vrai a chaque fois

```
$ PRODUCTION=true
$ [[ $PRODUCTION ]] && echo toto
toto
$ PRODUCTION=false
$ [[ $PRODUCTION ]] && echo toto
toto
$ [[ $PRODUCTION == false ]] && echo toto
toto
$ PRODUCTION=true
$ [[ $PRODUCTION == false ]] && echo toto
$ 
```

edgd1er committed:
ajout du supervisor pour docker / gestion production - dev

edgd1er committed:
removed unneeded var

edgd1er committed:
fix bad location for supervisor conf

edgd1er committed:
fix typo

edgd1er committed:
create missing directories first

edgd1er committed:
$? catch bash return values

Astral0 commented:
OK avec toi pour les [[ ]] on peut les changer en [ ].

Pour mysql/secret je pense qu'il ne sert pas.

De manière générale, je ne comprends pas la logique de faire "source toto.sh" et d'appeler les méthodes une à une.  Vu qu'on source 3 ou 4 fichiers, on se met en risque (mélange de fonctions et variables).

Perso je pense qu'on peut faire beaucoup plus simple : 
- un install/postinst qui est lancé par tous les utilisateurs d'une version dev (git clone + ./install/postinst). Il est lancé aussi par le paquet deb.
- le fichier postinst commence par déterminer si c'est une vdev ou apt, voir s'il est dans Docker (si c'est possible de le déterminer, ca sera plus simple).
- si vdev, alors postinst lance presinst, fait ses manips, et continue ensuite la procédure postinst
- si Docker, alors il lance *jesaispasquoi*, peut être un "preinst_docker" qui lance (ou pas) preinst, puis ensuite, postinst continue.

Je trouve cette multiplication des fichiers bash contre-productive, et avoir plusieurs points d'entrée c'est encore pire.. C'est clair qu'on va perdre les utilisateurs.
Dans tous les cas, on devrait juste demander à l'utlisateur de lancer "./install/postinst".

Je rappelle que c'est comme ca qu'est construit actuellement le fichier postinst, et qu'on avait dit qu'on éclatait juste ce fichier pour sortir ce qui n'est à faire que pour la vdev (et docker). On n'a jamais parlé du fait que l'utilisateur devait lancer un script ou un autre selon son contexte.

AA

Astral0 commented:
Cette partie du postinst.sh n'a rien à faire là (elle n'était affichée que dans le cas d'une vdev) :

```
cat - <<EOS
  Installation dir  : ${ROOT_DIRECTORY}
  Database host     : ${MYSQL_HOSTNAME}
  Database port     : ${MYSQL_PORT}
  Database name     : ${MYSQL_NEXTDOM_DB}
  Database user     : ${MYSQL_NEXTDOM_USER}
  Database password : ${MYSQL_NEXTDOM_PASSWD}
  > A self-signed SSL Certificate created in /etc/nextdom/ssl/
  > Please feel free to use another Certificate
  >>>>> COMPLETED <<<<<
EOS
```

edgd1er commented:
@Astral0 respires un grand coup ça ira mieux :)

L'idée initiale de @slobberbone  est de découper l'installation et la configuration en étapes. Ces étapes sont réparties dans deux fichiers selon le type d'installation ( git dev, deb, docker ). 

par exemple, une installation deb ne va avoir besoin que du postinst, une installation git et dev va avoir besoin du pre et du post. Voila qui devrait te rassurer quand a la multiplication des bash. @slobberbone a juste exporter la configuration et les fonctions communes dans deux fichiers bien nommés: config.sh et utils.sh. Ces deux fichiers sont appelés par les scripts les nécéssitant, donc c'est transparent pour l'utilisateur. Et puis ne pas prendre l'utilisateur pour un idiot mais lui indiquer comment faire dans un README fait partie des bonnes pratiques aussi. On peut le guider ;)

Le découpage actuel pré/post inst est adapté pour un installation de dev ce que je ne remets pas en cause. J'ai adapté le install_docker.sh via des sources pour ne prendre que les étapes utiles et nécéssaires à l'installation docker. Justement, je prends à mon compte la complexité pour changer le moins possible le pre et postinst, afin que les deux scripts restent utilisables tel que @slobberbone les a définit. C'est pour ça que les opérations des étapes ne doivent concerner qu'un sujet ( pas comme l'étape 6 du post qui mélange des étapes de l'installation modification du temp et des étapes de l'execution, modification des variables et injection du schema dans la bdd ). 

Je me repete, le but est bien de conserver le fonctionnement actuel:
* deb + postinst.sh
* git clone + pre-inst + postinst
* dockerfile + qq fct pre-inst + qq fct postinst

@slobberbone si j'ai mal compris tes intentions dans cette PR, n'hésites pas à me corriger.

pour info, dans mon dockerfile j'utilise la version buster ( balenalib/amd64-debian:buster-run), si vous jeter un oeil c'est la: https://github.com/NextDom/nextdom-docker.git dans la branche refactorPostInst. La construction se passe en trois étapes
* build d'une image de base
* build d'une image de dev depuis l'image de base et generation des fichiers ( css, js, themes, ....)
* build d'une image de prod depuis l'image de base avec copie des fichers générés. publication de cette image.
L'image produite est loin d'être legere ( 900Mb) mais la version une seule etape etait pire (1.3Gb).

slobberbone committed:
Some fixes

- fix for tests
- check log folder if exists
- add gen_composer.sh execution
- review [[ ]] to [ ]

edgd1er committed:
allow supervisor to kill spawned apache process

edgd1er committed:
fix always true tests, missing dirs

Sylvaner committed:
Fix tests

slobberbone commented:
@edgd1er peux-tu te connecter à Discord stp Je revois ton install_docker pour le mettre au propre

slobberbone committed:
Add composer, npm and assets

Gis70 reviewed (approved):
Fresh install debian 10, install_git OK merci

Gis70 reviewed (changes requested):
Il y a un pb de restitution du mdp DB : 
![image](https://user-images.githubusercontent.com/16589132/72763449-1cccfd80-3be4-11ea-9375-84a92bb34847.png)

vinceg77 commented:
Installe ok mais pb de mot de passe (cf ci dessus)

vinceg77 reviewed (changes requested):
Installe ok mais pb de mot de passe sur la page OS/DB (cf ci dessus)

Sylvaner committed:
Check BDD server


Remove docker option


Fix tests


Fix test

Sylvaner committed:
Fix squash

Sylvaner committed:
Fix travis again

Esso1-jeedom commented:
j'ai une question normale qu'un rep html soir recréé sous /var/www/html/html
![image](https://user-images.githubusercontent.com/40739563/73140892-3061e380-407e-11ea-9619-94171ce85697.png)
sinon c parfait

Esso1-jeedom commented:
debian 10 
armbian 10
Sources dans /var/www/nextdomdev 

Installation par Git + paquets debian : nextdom-common  + changement de branche.

Installation par Git + paquets debian : nextdom-common  
git clone https://github.com/NextDom/nextdom-core.git -b feature/RefactorPostinst . 
![image](https://user-images.githubusercontent.com/40739563/73186578-432cf480-4120-11ea-8c8e-e736d0c18340.png)

slobberbone committed:
Fix Secret_key trouble

Esso1-jeedom commented:
bon tjrs la merde sur fresh install.
manque tjrs le rep 
/usr/share/nextdom 
du coup ca marche pas 
en créer le rep 
j'ai le truc qu'on a pas le droit .
![image](https://user-images.githubusercontent.com/40739563/73385195-06523080-42cd-11ea-9b34-cd72118f825d.png)
quand on sup le nextdom_common
![image](https://user-images.githubusercontent.com/40739563/73386548-4f0ae900-42cf-11ea-89ae-52ff3fce5078.png)
et par magie ...
![image](https://user-images.githubusercontent.com/40739563/73386583-5cc06e80-42cf-11ea-8455-494bfd205a68.png)
(Reading database ... 52458 files and directories currently installed.)
Removing nextdom-common (1.2.0) ...
Enabling site 000-default.
To activate the new configuration, you need to run:
  systemctl reload apache2

je peux le reinstaller, avec apt install -y nextdom-common , c bon . tout fonctionne

Gis70 reviewed (changes requested):
Sur installation vm Buster, en install_git dans /usr/share/nextdom on a un symlink dans /html qui pointe vers lui même.

slobberbone committed:
Fix nextdom.conf apache in preinst

Esso1-jeedom commented:
peut on avoir la procedure d'install car moi cela ne marche pas  . 

wget -qO -   http://debian.nextdom.org/debian/nextdom.gpg.key  | apt-key add -
echo "deb   http://debian.nextdom.org/debian   nextdom main" >/etc/apt/sources.list.d/nextdom.list
apt update
apt -y install nextdom-common
# 
# Créer un répertoire pour y mettre les sources. Ce répertoire doit être accessible par l'utilisateur www-data donc ne pas le mettre dans /root/
mkdir /var/www/nextdomdev && cd /var/www/nextdomdev  
git clone https://github.com/NextDom/nextdom-core.git -b feature/RefactorPostinst .
./install/install_git.sh


root@orangepiplus:/var/www/nextdomdev/install# ./install_git.sh
02/01/20 09:19:37 AM STEP : Preinst -- Prepare and check prerequisites - 0/7
02/01/20 09:19:37 AM INFO : Local needed directories checked
02/01/20 09:19:37 AM INFO : Local mysql server detected
02/01/20 09:19:37 AM INFO : MySQL/MariaDB service started
/etc/php/7.3
02/01/20 09:19:37 AM INFO : PHP detected
02/01/20 09:19:37 AM INFO : Apache2 service started
02/01/20 09:19:37 AM INFO : Cron service started
02/01/20 09:19:37 AM SUCCESS : Prerequisites are checked with success
02/01/20 09:19:37 AM SCRIPT : ============ Starting preinst.sh ============
02/01/20 09:19:37 AM STEP : Preinst -- Generate Assets - 1/7
 >>> Installation dependencies composer

Fatal error: Uncaught Error: Call to undefined function Composer\Json\json_decode() in /usr/share/php/Composer/Json/JsonFile.php:153
Stack trace:
#0 /usr/share/php/Composer/Factory.php(290): Composer\Json\JsonFile->validateSchema(1)
#1 /usr/share/php/Composer/Factory.php(576): Composer\Factory->createComposer(Object(Composer\IO\ConsoleIO), './composer.json', false)
#2 /usr/share/php/Composer/Console/Application.php(345): Composer\Factory::create(Object(Composer\IO\ConsoleIO), NULL, false)
#3 /usr/share/php/Composer/Console/Application.php(458): Composer\Console\Application->getComposer(false, false)
#4 /usr/share/php/Composer/Console/Application.php(156): Composer\Console\Application->getPluginCommands()
#5 /usr/share/php/Symfony/Component/Console/Application.php(148): Composer\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput),                                              Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 /usr/share/php/Composer/Console/Application.php(104): Symfony\Component\Console\Application->run(Object(Symfony in /usr/share/php/Composer/Json/JsonF                                             ile.php on line 153
02/01/20 09:19:37 AM ERROR : error during composer and npm initialize
 > Initialise NPM
 >>> Installation dependencies composer

Fatal error: Uncaught Error: Call to undefined function Composer\Json\json_decode() in /usr/share/php/Composer/Json/JsonFile.php:153
Stack trace:
#0 /usr/share/php/Composer/Factory.php(290): Composer\Json\JsonFile->validateSchema(1)
#1 /usr/share/php/Composer/Factory.php(576): Composer\Factory->createComposer(Object(Composer\IO\ConsoleIO), './composer.json', false)
#2 /usr/share/php/Composer/Console/Application.php(345): Composer\Factory::create(Object(Composer\IO\ConsoleIO), NULL, false)
#3 /usr/share/php/Composer/Console/Application.php(458): Composer\Console\Application->getComposer(false, false)
#4 /usr/share/php/Composer/Console/Application.php(156): Composer\Console\Application->getPluginCommands()
#5 /usr/share/php/Symfony/Component/Console/Application.php(148): Composer\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput),                                              Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 /usr/share/php/Composer/Console/Application.php(104): Symfony\Component\Console\Application->run(Object(Symfony in /usr/share/php/Composer/Json/JsonF                                             ile.php on line 153
02/01/20 09:19:37 AM ERROR : error during asset generation
02/01/20 09:19:37 AM INFO : installed nodejs
02/01/20 09:19:37 AM INFO : installed composer manager
02/01/20 09:19:37 AM INFO : installed project dependencies
02/01/20 09:19:37 AM INFO : copied icons, themes and images from assets
02/01/20 09:19:37 AM INFO : generated css files
02/01/20 09:19:37 AM INFO : generated javascript files
02/01/20 09:19:37 AM STEP : Preinst -- Configure MySQL/MariaDB - 2/7
02/01/20 09:19:37 AM INFO : created nextdom mysql configuration: /etc/mysql/conf.d/nextdom_my.cnf
02/01/20 09:19:37 AM SUCCESS : MySQL is configured with success
02/01/20 09:19:37 AM STEP : Preinst -- Prepare /var/www/html- 3/7
02/01/20 09:19:37 AM INFO : warning : directory /var/www/html isn't empty, renamed to /var/www/html.fgfrw8Km
02/01/20 09:19:37 AM SUCCESS : /var/www/html is prepared with success
02/01/20 09:19:37 AM STEP : Preinst -- Configure Apache - 4/7
02/01/20 09:19:37 AM INFO : updated wwwdir in : /etc/apache2/sites-available/nextdom.conf
02/01/20 09:19:37 AM INFO : updated wwwdir in : /etc/apache2/sites-available/nextdom-ssl.conf
02/01/20 09:19:37 AM INFO : updated wwwdir in : /etc/apache2/sites-available/nextdom-common
Generating RSA private key, 2048 bit long modulus (2 primes)
.....................................................................+++++
.................+++++
e is 65537 (0x010001)
Signature ok
subject=C = FR, ST = Paris, L = Paris, O = Global Security, OU = IT Department, CN = example.com
Getting Private key
02/01/20 09:19:37 AM INFO : created SSL self-signed certificates in /etc/nextdom/ssl/
02/01/20 09:19:37 AM SUCCESS : Apache is configured with success
02/01/20 09:19:37 AM STEP : Preinst -- Configure MySQL/MariaDB - 5/7
02/01/20 09:19:37 AM INFO : Writing MariaDb/MySQL information file: /etc/nextdom/mysql/secret
02/01/20 09:19:37 AM SUCCESS : MySQL/MariaDB is configured with success
02/01/20 09:19:37 AM STEP : Preinst -- Generate MySQL/MariaDB structure - 6/7
02/01/20 09:19:37 AM INFO : deleted mysql user: nextdom
02/01/20 09:19:37 AM INFO : created mysql user: nextdom
02/01/20 09:19:37 AM INFO : deleted mysql table: nextdom
02/01/20 09:19:37 AM INFO : created mysql table: nextdom
02/01/20 09:19:37 AM INFO : configured table privileges: nextdom
02/01/20 09:19:37 AM SUCCESS : Database structure generated with success
02/01/20 09:19:37 AM STEP : Preinst -- Configure PHP - 7/7
02/01/20 09:19:37 AM INFO : Directory or file : /var/www/nextdomdev/assets/config/default.config.ini removed
02/01/20 09:19:37 AM INFO : Directory or file : /etc/php/7.3/apache2/conf.d/10-opcache.ini removed
02/01/20 09:19:37 AM INFO : production mode
02/01/20 09:19:37 AM INFO : enable PHP opcache
02/01/20 09:19:37 AM INFO : restart Apache
02/01/20 09:19:37 AM SUCCESS : PHP is configured with success
02/01/20 09:19:37 AM SCRIPT : ============ Preinst.sh is executed ... ============
02/01/20 09:20:03 AM SCRIPT : ============ Starting postinst.sh ============
02/01/20 09:20:03 AM INFO : Apache2 service started
02/01/20 09:20:03 AM INFO : Cron service started
02/01/20 09:20:03 AM INFO : Local mysql server detected
02/01/20 09:20:03 AM INFO : MySQL/MariaDB service started
02/01/20 09:20:03 AM STEP : Postinst -- Create needed files and directories - 1/12
02/01/20 09:20:03 AM INFO : Directory : /var/www/nextdomdev/plugins created
02/01/20 09:20:03 AM INFO : Directory or file : /var/log/nextdom/install.log removed
02/01/20 09:20:03 AM INFO : Directory or file : /tmp/jeedom removed
ln: failed to access '/tmp/jeedom': Permission denied
02/01/20 09:20:03 AM INFO : created temporary directory: /tmp/nextdom
02/01/20 09:20:03 AM SUCCESS : Files and directories are created with success
02/01/20 09:20:03 AM STEP : Postinst -- Prepare directory layout - 2/12
02/01/20 09:20:03 AM INFO : Directory or file : /var/lib/nextdom/config removed
02/01/20 09:20:03 AM INFO : created configuration directory /var/lib/nextdom/config
02/01/20 09:20:03 AM INFO : Directory or file : /var/lib/nextdom/data removed
02/01/20 09:20:03 AM INFO : created data directory /var/lib/nextdom/data
02/01/20 09:20:03 AM INFO : created core configuration symlink: /var/www/nextdomdev/core/config
02/01/20 09:20:03 AM INFO : created var symlink: /var/www/nextdomdev/var
02/01/20 09:20:03 AM INFO : created core/css symlink: /var/www/nextdomdev/core/css
02/01/20 09:20:03 AM INFO : created core/js symlink: /var/www/nextdomdev/assets/core/js
02/01/20 09:20:03 AM INFO : created core/template symlink: /var/www/nextdomdev/core/template
02/01/20 09:20:03 AM INFO : created data symlink: /var/www/nextdomdev/data
02/01/20 09:20:03 AM INFO : Directory or file : /var/www/nextdomdev/log removed
02/01/20 09:20:03 AM INFO : created log symlink: /var/www/nextdomdev/log
02/01/20 09:20:03 AM INFO : cache cleared
02/01/20 09:20:03 AM SUCCESS : NextDom is configured with success
02/01/20 09:20:03 AM STEP : Postinst -- Configure MySQL/MariaDB - 3/12
Uptime: 12  Threads: 7  Questions: 72  Slow queries: 0  Opens: 32  Flush tables: 1  Open tables: 26  Queries per second avg: 6.000
02/01/20 09:20:03 AM INFO : created nextdom mysql configuration: /etc/mysql/conf.d/nextdom_my.cnf
02/01/20 09:20:03 AM SUCCESS : MySQL is configured with success
02/01/20 09:20:03 AM STEP : Postinst -- Configure symbolic links - 4/12
02/01/20 09:20:03 AM SUCCESS : /var/www/nextdomdev linked with success to /var/www/html
02/01/20 09:20:03 AM STEP : Postinst -- Configure Apache - 5/12
Considering dependency setenvif for ssl:
Module setenvif already enabled
Considering dependency mime for ssl:
Module mime already enabled
Considering dependency socache_shmcb for ssl:
Enabling module socache_shmcb.
Enabling module ssl.
See /usr/share/doc/apache2/README.Debian.gz on how to configure SSL and create self-signed certificates.
To activate the new configuration, you need to run:
  systemctl restart apache2
02/01/20 09:20:03 AM INFO : apache: enable module ssl
Enabling module rewrite.
To activate the new configuration, you need to run:
  systemctl restart apache2
02/01/20 09:20:03 AM INFO : apache: enable module rewrite
Module status disabled.
To activate the new configuration, you need to run:
  systemctl restart apache2
02/01/20 09:20:03 AM INFO : apache: disable module status
Site 000-default already disabled
02/01/20 09:20:03 AM INFO : apache: disabled site default
Site default-ssl already disabled
02/01/20 09:20:03 AM INFO : apache: disabled site default-ssl
Enabling site nextdom-ssl.
To activate the new configuration, you need to run:
  systemctl reload apache2
02/01/20 09:20:03 AM INFO : apache: enabled site nextdom-ssl
Enabling site nextdom.
To activate the new configuration, you need to run:
  systemctl reload apache2
02/01/20 09:20:03 AM INFO : apache: enabled site nextdom
02/01/20 09:20:03 AM SUCCESS : Apache is configured with success
02/01/20 09:20:03 AM STEP : Postinst -- Configure NextDom - 6/12
02/01/20 09:20:03 AM INFO : wrote configuration file: /var/www/nextdomdev/core/config/common.config.php

Warning: require_once(/var/www/nextdomdev/vendor/autoload.php): failed to open stream: No such file or directory in /var/www/nextdomdev/src/core.php on                                              line 60

Fatal error: require_once(): Failed opening required '/var/www/nextdomdev/vendor/autoload.php' (include_path='.:/usr/share/php') in /var/www/nextdomdev/                                             src/core.php on line 60
02/01/20 09:20:03 AM ERROR : NextDom installation script failed
02/01/20 09:20:03 AM INFO : added user as sudoer: www-data
02/01/20 09:20:03 AM STEP : Postinst -- Restart MySQL/MariaDB - 7/12
02/01/20 09:20:03 AM SUCCESS : MySQL/MariaDB is configured with success
02/01/20 09:20:03 AM STEP : Postinst -- Configure Cron - 8/12
02/01/20 09:20:03 AM INFO : created nextdom cron configuration: /etc/cron.d/nextdom
02/01/20 09:20:03 AM INFO : created nextdom cron configuration: /etc/cron.d/nextdom_watchdog
[warn] /etc/environment has been deprecated for locale information; use /etc/default/locale for LC_ALL="C" instead ... (warning).
[ ok ] Reloading configuration files for periodic command scheduler: cron.
02/01/20 09:20:03 AM SUCCESS : Cron is configured with success
02/01/20 09:20:03 AM STEP : Postinst -- Check NextDom - 9/12
==================================================
|    NextDom SICK SCRIPT 2020-02-01 09:20:23    |
==================================================

**************************************************
*                 VARIABLES                      *
**************************************************

Fatal error: Uncaught Error: Call to undefined function posix_getpwuid() in /var/www/nextdomdev/scripts/sick.php:50
Stack trace:
#0 {main}
  thrown in /var/www/nextdomdev/scripts/sick.php on line 50
02/01/20 09:20:03 AM ERROR : Error while checking nextdom
02/01/20 09:20:03 AM STEP : Postinst -- Execute scripts for specific OS- 10/12
02/01/20 09:20:03 AM SUCCESS : OS specific actions are done with success
02/01/20 09:20:03 AM STEP : Postinst -- Configure file permission - 11/12
02/01/20 09:20:03 AM INFO : set file owner: www-data, perms: 0755/0644 on directory /var/lib/nextdom
02/01/20 09:20:03 AM INFO : set file owner: www-data, perms: 0755/0644 on directory /var/log/nextdom
02/01/20 09:20:03 AM INFO : set file owner: www-data, perms: 0755/0644 on directory /tmp/nextdom
02/01/20 09:20:03 AM INFO : set file owner: www-data, perms: 0755/0644 on directory /var/www/nextdomdev/plugins
02/01/20 09:20:03 AM INFO : set file owner: www-data, perms: 0755/0644 on directory /var/www/nextdomdev/public/img
02/01/20 09:20:03 AM SUCCESS : Files permissions are configured with success
02/01/20 09:20:03 AM STEP : Postinst -- Configure owner for NextDom directory - 12/12
02/01/20 09:20:03 AM SUCCESS : /var/www/nextdomdev, /var/lib/nextdom, /var/log/nextdom and /tmp/nextdom folder's owner is changed with success
02/01/20 09:20:03 AM SCRIPT : ============ Postinst.sh is executed ... ============
root@orangepiplus:/var/www/nextdomdev/install#

Esso1-jeedom commented:
donc c ok pour moi
Armbian buster 
debian Buster
je test cette aprem RPI 9 et 10
